test/torch/federated/test_dataset.py

```diff
@@ -1,10 +1,96 @@
-import pytest
 import torch as th
 import syft as sy
-import numpy as np
+
+from syft.frameworks.torch.federated import BaseDataset
+
+
+def test_base_dataset(workers):
+    bob = workers["bob"]
+
+    inputs = th.tensor([1, 2, 3, 4.0])
+    targets = th.tensor([1, 2, 3, 4.0])
+    dataset = BaseDataset(inputs, targets)
+    assert len(dataset) == 4
+    assert dataset[2] == (3, 3)
+
+    dataset.send(bob)
+    assert dataset.data.location.id == "bob"
+    assert dataset.targets.location.id == "bob"
+    assert dataset.location.id == "bob"
+
+    dataset.get()
+    try:
+        assert dataset.data.location.id == 0
+        assert dataset.targets.location.id == 0
+        assert False
+    except AttributeError:
+        pass
 
 
 def test_federated_dataset(workers):
```

Comment:
Federated class?

Reply:
Sorry?

Reply:
Should have been more clear. All these tests important base dataset. Should they he using federated dataset?

Reply:
I'm not sure I get all.
Basically this tests the BaseDataset class, which is supposed to be a class containing two tensors (.data and .targets) and which you can manipulate (.send(bob), .get() ) just like a tensor

Reply:
This is what to give to feed a FederatedDataset